benchmark/benchmark_test.go

```diff
@@ -273,3 +274,49 @@ func BenchmarkFuncrJSONLogInfoMarshalerValue(b *testing.B) {
 	var log logr.Logger = funcr.NewJSON(noopJSON, funcr.Options{})
 	doMarshalerValue(b, log)
 }
+
+var loggerResult logr.Logger
+var errResult error
+
+func BenchmarkFromContext(b *testing.B) {
+	var logger logr.Logger
+	var err error
+	ctx := context.Background()
+
+	b.Run("no-logger", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			logger, err = logr.FromContext(ctx)
+		}
+	})
+	loggerResult = logger
+	errResult = err
+
+	b.Run("with-logger", func(b *testing.B) {
+		ctx := logr.NewContext(ctx, logr.Discard())
+		for i := 0; i < b.N; i++ {
+			logger, err = logr.FromContext(ctx)
+		}
+	})
+	loggerResult = logger
+	errResult = err
+}
+
+func BenchmarkFromContextOrDiscard(b *testing.B) {
+	var logger logr.Logger
+	ctx := context.Background()
+
+	b.Run("no-logger", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
```

Comment:
We should probably add `b.ReportAllocs()` to all of these?  Maybe a new commit before this series.

Reply:
I typically run with `-benchmem` when I want allocations. I'm not sure whether I want benchmarks to override that choice - that's what `b.ReportAllocs()` does, doesn't it?

Reply:
Oh, somehow I didn't know that and just always did it manually.  Ha!